source/module_hsolver/hsolver_lcao.cpp

```diff
@@ -161,12 +221,132 @@ void HSolverLCAO<T>::solve(hamilt::Hamilt<T>* pHamilt,
     this->solveTemplate(pHamilt, psi, pes, this->method, skip_charge);
 }
 
-template <typename T>
-void HSolverLCAO<T>::hamiltSolvePsiK(hamilt::Hamilt<T>* hm, psi::Psi<T>& psi, double* eigenvalue)
+template <typename T, typename Device>
+void HSolverLCAO<T, Device>::hamiltSolvePsiK(hamilt::Hamilt<T>* hm, psi::Psi<T>& psi, double* eigenvalue)
 {
     ModuleBase::TITLE("HSolverLCAO", "hamiltSolvePsiK");
     ModuleBase::timer::tick("HSolverLCAO", "hamiltSolvePsiK");
-    this->pdiagh->diag(hm, psi, eigenvalue);
+
+    if (this->method != "cg_in_lcao")
+    {
+        this->pdiagh->diag(hm, psi, eigenvalue);
+    }
+    else 
```

Comment:
Not good to me, This code block should not place in functions in HSolverLCAO class!